### ISSUE TITLE
`norm='batch'` in UNet cause int64 weights

### ISSUE THREAD

**Issue by Nic-Ma** [Project-MONAI/MONAI#2155]
**Describe the bug**
When set the norm of UNet to `batch`, it will cause "model.0.conv.unit0.conv.weight" to be float64 data.
While `instance` norm is fine.

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

Could you please help confirm this issue and see whether we can convert it to `float32`?

Thanks in advance.

**Comment by ericspod** [comment]
All weights should be `float32` as far as I know. I don't see how this became `float64`, we ultimately instantiate one of the Pytorch batchnorm layers.

**Comment by holgerroth** [comment]
@Nic-Ma It's causing int64 weights.

**Comment by Nic-Ma** [comment]
Hi @holgerroth ,

Could you please help provide a simple program to reproduce the issue?
Maybe print the weight type of that layer when using `batch norm`?

Thanks in advance.

**Comment by holgerroth** [comment]
Batchnorm seems to create LongTensors which causes the issue. See this simple test script comparing UNet with instance norm vs. batch norm:

`#!/usr/bin/env python3
import numpy as np
from monai.networks.nets import UNet


def count_long_tensors(net):
    state_dict = net.state_dict()

    n_long_tensors = 0
    for key in state_dict.keys():
        if state_dict[key].type() == "torch.LongTensor":
            print(f"{key}: {state_dict[key].type()}, {np.shape(state_dict[key])}")
            n_long_tensors += 1

    return n_long_tensors


unet_instance = UNet(dimensions=3,
                       in_channels=1,
                       out_channels=2,
                       channels=[16, 32, 64, 128, 256],
                       strides=[2, 2, 2, 2],
                       num_res_units=2)
unet_batch = UNet(dimensions=3,
                   in_channels=1,
                   out_channels=2,
                   channels=[16, 32, 64, 128, 256],
                   strides=[2, 2, 2, 2],
                   num_res_units=2,
                   norm="batch")

print("Long tensors:")
print("Instance Norm Unet", count_long_tensors(unet_instance))
print("Batch Norm Unet", count_long_tensors(unet_batch))
`

**Output:**
`Long tensors:
Instance Norm Unet 0
model.0.conv.unit0.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.0.conv.unit1.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.0.conv.unit0.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.0.conv.unit1.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.1.submodule.0.conv.unit0.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.1.submodule.0.conv.unit1.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.1.submodule.1.submodule.0.conv.unit0.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.1.submodule.1.submodule.0.conv.unit1.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.1.submodule.1.submodule.1.submodule.conv.unit0.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.1.submodule.1.submodule.1.submodule.conv.unit1.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.1.submodule.1.submodule.2.0.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.1.submodule.1.submodule.2.1.conv.unit0.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.1.submodule.2.0.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.1.submodule.2.1.conv.unit0.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.2.0.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.1.submodule.2.1.conv.unit0.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
model.2.0.adn.N.num_batches_tracked: torch.LongTensor, torch.Size([])
Batch Norm Unet 17`

**Comment by Nic-Ma** [comment]
Hi @holgerroth ,

Thanks for providing the test program.
I checked your log print, seems all the logs are related to this int number: `adn.N.num_batches_tracked`.
https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/blocks/acti_norm.py#L20

Thanks.

**Comment by Nic-Ma** [comment]
Hi @holgerroth ,

Actually, this int64 variable comes from PyTorch source code of batch norm:
https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/batchnorm.py#L54
You can removed by below settings in your UNet args:
```py
unet_batch = UNet(
    dimensions=3,
    in_channels=1,
    out_channels=2,
    channels=[16, 32, 64, 128, 256],
    strides=[2, 2, 2, 2],
    num_res_units=2,
    norm=("batch", {"track_running_stats": False}),
)
```
I already verified locally, will also enhance our unit tests to cover it.
Could you please help double confirm?

Thanks.
